src/python/pants/help/help_info_extracter.py

```diff
@@ -246,10 +248,22 @@ class RuleInfo:
 class AllHelpInfo:
     """All available help info."""
 
-    scope_to_help_info: dict[str, OptionScopeHelpInfo]
-    name_to_goal_info: dict[str, GoalHelpInfo]
-    name_to_target_type_info: dict[str, TargetTypeHelpInfo]
-    rule_output_type_to_rule_infos: dict[str, tuple[RuleInfo, ...]]
+    scope_to_help_info: LazyFrozenDict[str, OptionScopeHelpInfo]
+    name_to_goal_info: LazyFrozenDict[str, GoalHelpInfo]
+    name_to_target_type_info: LazyFrozenDict[str, TargetTypeHelpInfo]
+    rule_output_type_to_rule_infos: LazyFrozenDict[str, tuple[RuleInfo, ...]]
+
+    def asdict(self) -> dict[str, Any]:
+        return {
+            field: {
+                # This will be simplified when the rule infos are flattened...
+                thing: tuple(dataclasses.asdict(t) for t in info)
+                if isinstance(info, tuple)
+                else dataclasses.asdict(info)
```

Comment:
You can make this a lot more readable via Black if you add `()` around the if/else expression, it will indent things properly.